Formula/s/sf.rb

```diff
@@ -0,0 +1,21 @@
+class Sf < Formula
+  desc "Command-line toolkit for Salesforce development"
+  homepage "https://developer.salesforce.com/tools/salesforcecli"
+  url "https://registry.npmjs.org/@salesforce/cli/-/cli-2.61.8.tgz"
+  sha256 "6d7388565f83c15edd2ad05d1f42737de64680b8ea18a7f569435e93e00d6620"
+
+  license "BSD-3-Clause"
+  # See https://github.com/salesforcecli/cli
+
+  depends_on "node@20"
+  # Depends on v20 to avoid the `punycode` warnings
+
+  def install
+    system "npm", "install", *std_npm_args
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/sf --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate tebump-git-branchless-0.10.0ath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Added tests that create an empty project and test the output.

```ruby
  test do
    system "#{bin}/sf project generate -n projectname -t empty"
    assert_predicate testpath/"projectname", :exist?
    assert_predicate testpath/"projectname/config/project-scratch-def.json", :exist?
    assert_predicate testpath/"projectname/README.md", :exist?
    assert_predicate testpath/"projectname/sfdx-project.json", :exist?
    assert_predicate testpath/"projectname/.forceignore", :exist?
  end
```